### ISSUE TITLE
Apple Notarization

### ISSUE THREAD

**Issue by simongregory** [castlabs/electron-releases#38]
To prepare for macOS 10.15 I've worked through Apple's [notarization](https://developer.apple.com/documentation/security/notarizing_your_app_before_distribution) and got our app (v5.0.5-wvvmp) working for hardened runtime (although not actually tested on macOS 10.15 yet). This boiled down working out what [entitlement](https://developer.apple.com/documentation/security/hardened_runtime_entitlements) to add. Using:

```
<key>com.apple.security.cs.disable-library-validation</key>
true/>
```

seemed to be enough.

Originally the expectation was that the app would just crash if it there was a violation. Without the above entitlement it didn't. What would happen is that cdm was reported as loaded but then accessing it threw an error. 

This leaves me wondering about stability, and that future releases of the CDM might require alternative entitlements. When they're upgraded to the results could be problematic. Or it could be that I'm denying the current cdm access to something it needs and the right circumstances haven't arisen for me to see it error.

So it seemed sensible to ask if there's any known notarisation issues? I imagine that if there are plans for the CDM's to also be signed/notarized then we won't have to worry?

**Comment by khwaaj** [comment]
I do not personally know of any additional notarization issues, at least not currently, but since it is not something I've been able to test myself so far I cannot say with complete certainty. Neither do I currently know of any plans to sign/notarize the CDM itself. I'll bring this up at my next sync with the Widevine team and see if they have any input.

**Comment by khwaaj** [comment]
Apparently the 4.10.1440.18 release of the CDM was not Apple code-signed by Google, which would cause issues. This was fixed in 4.10.1440.19 which was released last week and it is now Apple code-signed by Google (as well as VMP signed, of course). Electron should automatically update to it, so please use that for any further testing (if you are not already).

This fix does not lift the requirement on using the `com.apple.security.cs.disable-library-validation` entitlement since the CDM library is singled by another developer, however my contacts mentioned no other entitlements necessary for the CDM to work.

**Comment by simongregory** [comment]
Great, thanks for the info @khwaaj. 

Sounds like there's a very small chance of problems on CDM upgrade. However I'm going to access the cdm shortly after you declare it ready and double check it's working. If it's not then I can at least fire some monitoring events so we're aware of issues.

We got a beta install of 10.15 and the app works fine. All the notarization dialogs etc don't work as expected, but this appears to be Apples fault as there's a few posts in the Apple dev forums reporting similar behaviour from other devs.
